desktop/libs/indexer/src/indexer/templates/importer.mako

```diff
@@ -2827,6 +2834,14 @@ ${ commonheader(_("Importer"), "indexer", user, request, "60px") | n,unicode }
       });
       self.readyToIndex = ko.pureComputed(function () {
         var validFields = self.destination.columns().length || self.destination.outputFormat() === 'database';
+        var isValidColumnNames = self.destination.columns().every(function (column) {
+          if (vm.sourceType === 'hive') {
```

Comment:
Why don't you use the self.columnNamePattern here to get the correct regex instead of duplicating the logic and regexes? Something like:

`return RegExp(self.columnNamePattern()).test(column.name())`

Reply:
I tried test on the self.destination.columns() but It was giving error, and I was not aware of RegExp(). Thanks for the info.

Reply:
Ahh, ok looks like they are not in the same JS object. 
You could create a pure function in the JS root (somewhere close to the `getNewFieldName` function) and call that from both your columnNamePattern observable and from the readyToIndex observable.

```
var getColumnNameValidationRegex = function (sourceType) {
        if (sourceType === 'hive') {
          return "[a-zA-Z0-9_ ]+$";
        } else {
          return "[a-zA-Z0-9_]+$";
        }
    };
```

Reply:
`RegExp(self.columnNamePattern()).test(column.name())` Your suggestion is working fine so I think we don't need the new function.